analyzer/src/main/kotlin/managers/Conan.kt

```diff
@@ -0,0 +1,316 @@
+                        dependencies = extractDependencyTree(rootNode, workingDir, child, SCOPE_NAME_DEV_DEPENDENCIES)
+                    )
+                    result.add(packageDevReference)
+                }
+            }
+        }
+        return result.toSortedSet()
+    }
+
+    /**
+     * Runs through each package and extracts list of its dependencies (also transitive ones)
+     */
+    private fun extractDependencies(
+        rootNode: JsonNode,
+        scopeName: String,
+        workingDir: File
+    ): SortedSet<PackageReference> {
+        val stack = Stack<JsonNode>()
+        val dependencies = mutableSetOf<PackageReference>()
+
+        stack.addAll(rootNode)
+        while (!stack.empty()) {
+            val pkg = stack.pop()
+            extractDependencyTree(rootNode, workingDir, pkg, scopeName).forEach {
+                dependencies.add(it)
+            }
+        }
+        return dependencies.toSortedSet()
+    }
+
+    /**
+     * Extracts plain list of all packages
+     */
+    private fun extractPackages(node: List<JsonNode>, workingDir: File): Map<String, Package> {
+        val result = mutableMapOf<String, Package>()
+        val stack = Stack<JsonNode>()
+        stack.addAll(node)
+        while (!stack.empty()) {
+            val currentNode = stack.pop()
+            val pkg = extractPackage(currentNode, workingDir)
+            result["${pkg.id.name}:${pkg.id.version}"] = pkg
+        }
+        return result
+    }
+
+    /**
+     * Creates corresponding `Package` object from given `JsonNode` object
+     */
+    private fun extractPackage(node: JsonNode, workingDir: File) =
+        Package(
+            id = extractPackageId(node, workingDir),
+            declaredLicenses = extractDeclaredLicenses(node),
+            description = extractPackageField(node, workingDir, "description"),
+            homepageUrl = node["homepage"].textValueOrEmpty(),
+            binaryArtifact = RemoteArtifact.EMPTY, // TODO: implement me!
+            sourceArtifact = RemoteArtifact.EMPTY, // TODO: implement me!
+            vcs = VcsInfo(
+                type = VcsType.GIT,
+                url = node["url"].textValueOrEmpty(),
+                revision = node["revision"].textValueOrEmpty()
+            )
+        )
+
+    private fun runInspectRawField(pkgName: String, workingDir: File, field: String): String {
+        return run(workingDir, "inspect", pkgName, "--raw", field).stdout
+    }
+
+    private fun removeProjectPackage(rootNode: JsonNode, definitionFile: File): List<JsonNode> {
+        val projectPackage: JsonNode? = rootNode.find {
+            // Contains because conanfile.py's reference string often includes other data.
+            it["reference"].textValueOrEmpty().contains(definitionFile.name)
+        }
+        return if (projectPackage != null) {
+            rootNode.minusElement(projectPackage)
+        } else {
+            rootNode.toList<JsonNode>()
+        }
+    }
```

Comment:
More Kotlin-ish:

    private fun removeProjectPackage(rootNode: JsonNode, definitionFile: File): List<JsonNode> =
        rootNode.find {
            // Contains because conanfile.py's reference string often includes other data.
            it["reference"].textValueOrEmpty().contains(definitionFile.name)
        }?.let { projectPackage ->
            rootNode.minusElement(projectPackage)
        } ?: rootNode.toList<JsonNode>()